nemo/collections/llm/gpt/model/megatron/hyena/hyena_layer_specs.py

```diff
@@ -49,100 +57,82 @@ def _raise_te_import_error(*args, **kwargs):
     TERowParallelLinear = _raise_te_import_error
     TEDotProductAttention = _raise_te_import_error
 
-# Layer spec with TE modules
-if HAVE_TE:
-    hyena_stack_spec = ModuleSpec(
+
+def get_hyena_stack_spec(
+    use_te=HAVE_TE,
+    vortex_style_fp8=False,
+    unfused_rmsnorm=False,
+    plain_row_linear=False,
+):
+    """Construct desired Hyena stack spec based on given parameters."""
+    if use_te:
+        row_linear = TERowParallelLinear
+        col_linear = TELayerNormColumnParallelLinear
+        pre_layernorm = IdentityOp  # fused Norm+Linear, so no pre norm
+        core_attention = TEDotProductAttention
+    else:
+        row_linear = RowParallelLinear
+        col_linear = ColumnParallelLinear
+        pre_layernorm = TENorm  # would raise error if attempt to use
+        core_attention = DotProductAttention
+
+    if unfused_rmsnorm:
+        col_linear = RMSNormLinear
+
+    if vortex_style_fp8:
+        if unfused_rmsnorm:
+            dense_projection = RMSNormTELinearFp8
+        else:
+            dense_projection = TELayerNormColumnParallelLinearFp8
+    else:
+        dense_projection = col_linear
+
+    if plain_row_linear:
+        row_linear = Linear
```

Comment:
Is there a good place to raise an exception if the user asks for both `plain_row_linear` and tensor_parallelism? Maybe as part of `post_init` on the base hyena config?